app/src/components/utils.js

```diff
@@ -21,6 +21,22 @@ const utils = {
     else return str;
   },
 
+  /**
+   * @function convertObjectKeysToLowercase
+   * @param {object} obj an object with keys of mixed-case
+   * @returns the input object with keys converted to lowercase
+   */
+  convertObjectKeysToLowercase(obj){
+    const keys = Object.keys(obj);
+    let n = keys.length;
+    let newObj = {};
+    while (n--) {
+      let key = keys[n];
+      newObj[key.toLowerCase()] = obj[key];
+    }
+    return newObj;
+  },
+
```

Comment:
This can be written more idiomatically in declarative format:
```js
  /**
   * @function toLowerCaseKeys
   * @param {object} obj an arbitrary object
   * @returns {object} the object obj with all keys converted to lowercase
   */
  toLowerCaseKeys(obj) {
    return Object.fromEntries(Object.entries(obj).map(entry => {
      return [entry[0].toLowerCase(), entry[1]];
    })));
  }
```